subsys/bluetooth/controller/ll_sw/lll_chan.c

```diff
@@ -158,20 +194,305 @@ static uint16_t chan_mam(uint16_t a, uint16_t b)
 	return ((uint32_t)a * 17U + b) & 0xFFFF;
 }
 
-static uint16_t chan_prn(uint16_t counter, uint16_t chan_id)
+static uint16_t chan_prn_s(uint16_t counter, uint16_t chan_id)
 {
 	uint8_t iterate;
-	uint16_t prn_e;
+	uint16_t prn_s;
 
-	prn_e = counter ^ chan_id;
+	prn_s = counter ^ chan_id;
 
 	for (iterate = 0U; iterate < 3; iterate++) {
-		prn_e = chan_perm(prn_e);
-		prn_e = chan_mam(prn_e, chan_id);
+		prn_s = chan_perm(prn_s);
+		prn_s = chan_mam(prn_s, chan_id);
 	}
 
+	return prn_s;
+}
+
+static uint16_t chan_prn_e(uint16_t counter, uint16_t chan_id)
+{
+	uint16_t prn_e;
+
+	prn_e = chan_prn_s(counter, chan_id);
 	prn_e ^= chan_id;
 
 	return prn_e;
 }
+
+#if defined(CONFIG_BT_CTLR_ADV_ISO) || defined(CONFIG_BT_CTLR_SYNC_ISO)
+static uint8_t chan_sel_remap_index(uint8_t *chan_map, uint8_t chan_index)
+{
+	uint8_t octet_count;
+	uint8_t remap_index;
+
+	remap_index = 0U;
+	octet_count = 5U;
+	while (octet_count--) {
+		uint8_t octet;
+		uint8_t bit_count;
+
+		octet = *chan_map;
+		bit_count = 8U;
+		while (bit_count--) {
+			if (!chan_index) {
+				return remap_index;
+			}
+			chan_index--;
+
+			if (octet & BIT(0)) {
+				remap_index++;
+			}
+			octet >>= 1;
+		}
+
+		chan_map++;
+	}
+
+	return 0;
+}
+
+static uint16_t chan_prn_subevent_se(uint16_t chan_id,
+				     uint16_t *prn_subevent_lu)
+{
+	uint16_t prn_subevent_se;
+	uint16_t lu;
+
+	lu = *prn_subevent_lu;
+	lu = chan_perm(lu);
+	lu = chan_mam(lu, chan_id);
+
+	*prn_subevent_lu = lu;
+
+	prn_subevent_se = lu ^ chan_id;
+
+	return prn_subevent_se;
+}
+
+static uint8_t chan_d(uint8_t n)
+{
+	uint8_t x, y;
+
+	if (n > 5) {
+		x = n - 5;
+	} else {
+		x = 0;
+	}
+
+	if (n > 10) {
+		y = (n - 10) >> 1;
+	} else {
+		y = 0;
+	}
+
+	return MAX(1, MAX(MIN(3, x), MIN(11, y)));
+}
+#endif /* CONFIG_BT_CTLR_ADV_ISO || CONFIG_BT_CTLR_SYNC_ISO */
+
+#if defined(CONFIG_BT_CTLR_TEST)
+void lll_chan_sel_2_ut(void)
```

Comment:
It somewhat makes sense to have this as a single function, although I think there might be some readability to be gained by splitting it out. It seems that it's already logically separated using empty lines - I think separation by functions is a better solution.

Reply:
Tests are at best in the order close to what I could translate the sample data table in the BT Spec, table 3.1 and table 3.2. Refactoring will make it difficult to relate to the documented values and order of inputs used for testing.